docs/rules/func-names.md

```diff
@@ -45,6 +46,34 @@ Foo.prototype.bar = function bar() {};
 }())
 ```
 
+### as-needed
+
+In ES6, aside from explicit names, function names may be come from the surrounding context of the function. For example, a function assigned to a variable will automatically use the name of the variable as its name if it have no name of its own.
+
+Examples of **incorrect** code for this rule with the default `"as-needed"` option:
+
+```js
+/*eslint func-names: ["error", "as-needed"]*/
+
+Foo.prototype.bar = function() {};
```

Comment:
Since this doesn't get an inferred name, erroring on it seems like the right approach to me. Expanding the rule for the same of a V8-specific case seems less than ideal.